specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2016-09-01-preview/vmwarevirtustream.json

```diff
@@ -0,0 +1,1304 @@
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "tags": [
+    {
+      "name": "PrivateClouds",
+      "description": ""
+    },
+    {
+      "name": "Clusters",
+      "description": ""
+    },
+    {
+      "name": "Subscriptions",
+      "description": ""
+    },
+    {
+      "name": "Move Resource",
+      "description": ""
+    }
+  ],
+  "parameters": {
+    "subscriptionId": {
+      "name": "subscriptionId",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The unique identifier for a Microsoft Azure subscription."
+    },
+    "resourceGroupName": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the resource group within the Azure subscription.",
+      "x-ms-parameter-location": "method"
+    },
+    "apiVersion": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "description": "The Version of the API to be used with the client request."
+    }
+  },
+  "paths": {
+    "/providers/Microsoft.VMwareVirtustream/operations": {
+      "get": {
+        "tags": [
+          "Operations"
+        ],
+        "description": "Lists all of the available RP operations.",
+        "operationId": "Operations_List",
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Operations_List": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.VMwareVirtustream/privateClouds": {
+      "get": {
+        "operationId": "PrivateClouds_List",
+        "tags": [
+          "PrivateClouds"
+        ],
+        "summary": "list private clouds",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/PrivateCloudResponseList"
```

Comment:
This response array is wrapped in a "value" field, so I assume the goal is to adhere to the paging contract, but only ever return all the results on a single page. If so, please can you add the following declaration (and same for other pseudo-paged APIs):

```
"x-ms-pageable": {
  "nextLinkName": null
},
```

Reply:
I assume you mean `"nextLinkName": "nextLink"` like other APIs, not `null`.

Reply:
Ah, the reason I assumed null was because your PrivateCloudResponseList definition doesn't contain a "nextLink" field. If this API will implement paging then I think you just need to add "nextLink to the PrivateCloudResponseList definition, and as you mentioned, set "nextLinkName" (basically a pointer to the field to use for paging) to "nextLink".

Reply:
I added this here to PrivateCloud_List. `Cluster_List` is not currently pageable and it is not wrapped in a `value` field. I think it would be a good idea for the next API version.

Reply:
You are right. It doesn't contain a `nextLink` field, so I will set it to `null`.

Reply:
This breaks the build with error:
```
R3008 - CollectionObjectPropertiesNaming
Collection object 'PrivateCloudResponseList' returned by list operation 'PrivateClouds_List' with 'x-ms-pageable' extension, has no property named 'value'.
at specification/vmwarevirtustream/resource-manager/Microsoft.VMwareVirtustream/preview/2016-09-01-preview/vmwarevirtustream.json:141
```

Reply:
Let updated the wrong `nextLink`. Let me try again.

Reply:
This is working as you described after I added a `"type": "array"` in https://github.com/Azure/azure-rest-api-specs/pull/6653/commits/aeacc877abafe1a5122f1d72eff89f0d9d375a35